kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/vault/Vault.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.installation.vault;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.systemtests.Constants;
+import io.kroxylicious.systemtests.k8s.exception.KubeClusterException;
+import io.kroxylicious.systemtests.resources.manager.ResourceManager;
+import io.kroxylicious.systemtests.utils.DeploymentUtils;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+
+/**
+ * The type Vault.
+ */
+public class Vault {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Vault.class);
+    private static final String VAULT_CMD = "vault";
+    private final String deploymentNamespace;
+
+    /**
+     * Instantiates a new Vault.
+     *
+     * @param deploymentNamespace the deployment namespace
+     */
+    public Vault(String deploymentNamespace) {
+        this.deploymentNamespace = deploymentNamespace;
+    }
+
+    /**
+     * Is available
+     *
+     * @return true if Vault service is available in kubernetes, false otherwise
+     */
+    public boolean isAvailable() {
+        return kubeClient().getDeployment(deploymentNamespace, Constants.VAULT_SERVICE_NAME) != null;
+    }
+
+    /**
+     * Deploy.
+     *
+     */
+    public void deploy() {
+        LOGGER.info("Deploy HashiCorp Vault in {} namespace", deploymentNamespace);
+        if (isAvailable()) {
+            LOGGER.warn("Skipping Vault deployment. It is already deployed!");
+            return;
+        }
+
+        Map<String, String> values = new HashMap<>();
+        // server
+        values.put("server.dev.enabled", "true");
+        values.put("server.dev.devRootToken", Constants.VAULT_ROOT_TOKEN);
+        values.put("server.ha.enabled", "false");
+        values.put("server.updateStrategyType", "RollingUpdate");
+        values.put("server.service.type", "NodePort");
+        // injector
+        values.put("injector.enabled", "false");
```

Comment:
I can describe what I've done to make TLS work with dev mode in the integration tests. Testcontainers uses the upstream vault image.

Dev mode is useful, the server starts up unsealed so you don't have to do any work to initialise and unseal the vault. However, in dev mode you aren't in control of the main port configuration. It will start up a plain listener on 8200. You can configure additional ports though if you can add configuration files to the container. In the test I put:

```
listener "tcp" {
  address       = "0.0.0.0:8202"
  tls_cert_file      = "/vault/config/cert.pem"
  tls_key_file       = "/vault/config/key.pem"
}
```
in `/vault/config/Vault.hcl` to configure an additional TLS enabled listener on 8202 (there is also a property to configure the server trust for mutual TLS).

Configuring it this way with the Helm chart doesn't look so simple. There are conveniences for setting up the Vault config file if you are in [standalone](https://developer.hashicorp.com/vault/docs/platform/k8s/helm/configuration#config) or `ha` mode, but not in `dev` mode. Maybe you can mount in your own config file? Hashicorp have a [tutorial](https://developer.hashicorp.com/vault/tutorials/kubernetes/kubernetes-minikube-tls) using k8s to manage the certs but they run the vault in `ha` mode, so they can have custom config referencing their mounted in certificates (you can see the mucking around with unsealing required, they have instructions to exec in to each node and run unseal commands).